Assignment 2.Rmd

```diff
@@ -55,37 +52,47 @@ names(V.STD) <- c("id", "gender", "topic", "group")
 
 #Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
 VERTEX <- bind_rows(V.TCH, V.STD)
+VERTEX$gender <- as.factor(VERTEX$gender)
 
 ```
 
 Now we have both a Vertex and Edge list it is time to plot our graph!
 
 ```{r}
 #Load the igraph package
-
 library(igraph)
 
 #First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
-
 g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
 
 #Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
-
 plot(g,layout=layout.fruchterman.reingold)
 
 #There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
-
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 #We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
-In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+####In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+EDGE <- count(D2, tid, stid)
+EDGE.st <- EDGE%>%group_by(stid)%>%summarise(sum(n))
+EDGE.t <- EDGE%>%group_by(tid)%>%summarise(sum(n))
+names(EDGE.st) <- c("id", "count")
+names(EDGE.t) <- c("id", "count")
+EDGE.total <- bind_rows(EDGE.st,EDGE.t)
+VERTEX2 <- full_join(VERTEX,EDGE.total,by="id")
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX2$gender, vertex.size=EDGE.total$count)
+
```

Comment:
Well done! You clearly understand the basics of generating a graph with iGraph in R.